src/main/java/seedu/mtracker/filemanager/InstrumentDecoder.java

```diff
@@ -40,7 +41,11 @@ public static void setDoneStatus(boolean isDone, Instrument doneInstrument) {
         if (!isDone) {
             return;
         }
-        doneInstrument.markAsDone();
+        try {
+            doneInstrument.markAsDone();
+        } catch (AlreadyDoneError alreadyDoneError) {
+            alreadyDoneError.getMessage();
+        }
```

Comment:
Is there a reason why you have to do a check here in instrumentDecoder? InstrumentDecoder takes file data and convert it to instruments, it has nothing to do with the user setting the done status

Reply:
The ide shows an error if I don't handle the exception thrown by markAsDone method. That's why

Reply:
Then use the instrument doneInstrument method <- use this method to handle the exception. Its bad design to handle an exception that is not related to the class at all

Reply:
I don't think that's bad design at all. The handling of exceptions is related to the methods we are invoking, rather than the class in which this method is used.

I understand what you mean because in this case, this error will never occur due to the file setting the instrument as done, rather than an user. But, it is still acceptable to handle the error by virtue of the method being invoked.

On the contrary, I would say having a redundant doneInstrumentMethod() method inside InstrumentManager to achieve something which can easily be done through Instrument class itself is bad design. We are coupling both InsrtumentDecoder and DoneCommand classes to InstrumentManager unnecessarily.

Reply:
In case you forgot the instrumentManager is suppose to manage the instruments which means perform the actions on the instruments. 
The instrumentManager is already coupled with the classes despite your modification. Just because its in the abstract index class does not mean its not coupled with DoneCommand. But by doing this you are unnecessarily coupling an unrelated exception to the storage. Sry this is not negotiable

Reply:
Actually scrap my comment above, I think we shld ask Rafael he may know more

Reply:
actually maybe could incorporate this error into validate?

Reply:
> The instrumentManager is suppose to manage the instruments which means perform the actions on the instruments.

But done status is an attribute inside the instrument object though. So setting the done value should be done by a method inside instrument itself, no need another facade method from another class to just call only on this method once again. That is bad design tbh.

But I understand where you're coming from. The exception handling is unnecessary but we can't do without it since we are calling that method. But I do not think having this extra exception handling goes against any coding guidelines.

Reply:
Thanks @williamlamjy for your suggestion. We implemented it.